### PR TITLE
feat: HCP slot for home page

### PR DESCRIPTION
**Pull request by zchsh**
## 🔗 Relevant links

- [Preview link][preview] 🔎
- [Asana task][task] 🎟️

## 🗒️ What

Replaces the HashiConf CTA on the home page with an HCP CTA.

## 🤷 Why

To ensure the CTAs in our home page slots are relevant even as HashiConf is underway and then finished.

## 🛠️ How

- Adds a new `LogoCtaSlot`, as a base content-less presentation component
- Uses `LogoCtaSlot` to build `HcpSlot`
- Swaps out `HashiConfSlot` for `HcpSlot`
- Updates `VaultSlot` to use `LogoCtaSlot` under-the-hood for consistency

## 📸 Design Screenshots

🎨 [Figma][figma]

### Design Spec

CleanShot 2022-10-04 at 16 59 07@2x

### Before

before

### After

after

## 🧪 Testing

- [ ] Visit [the home page][preview]
- [ ] Ensure the new HCP slot looks and functions as expected, across all viewport sizes
- [ ] Ensure the Vault slot looks and functions as expected, across all viewport sizes
    - This component has been updated for consistency with the HCP slot

## 💭 Anything else?

I've updated the `VaultSlot` to use an abstracted `LogoCtaSlot`. This is likely not the ideal abstraction long-term, but felt useful in the short-term at least to ensure consistency in slot appearance:

> **Note**: these screenshots show a version of the HCP graphic that is from earlier today. The graphic has since been updated, see the `Design Screenshots` section above.

### Before slot consistency

2022-10-04-slot-consistency-before

### After slot consistency

2022-10-04-slot-consistency-after

[preview]: https://dev-portal-git-zsupdate-home-page-merch-slot-hashicorp.vercel.app
[task]: https://app.asana.com/0/1202097197789424/1203011399855189/f
[figma]: https://www.figma.com/file/VD7ahvXuXWJApeGnhbW4hv/Developer?node-id=14944%3A188600

### CODE REVIEW

File: src/content/home-page.json [hunk 1]
```diff
@@ -9,11 +9,14 @@
 		"vault": {
 			"cardTitle": "Vault",
 			"url": "/vault/tutorials/cloud/vault-introduction",
-			"description": "Access Vault’s secrets management and encryption capabilities instantly using the HCP portal",
+			"description": "Access Vault’s secrets management and encryption capabilities instantly using the HCP portal.",
```
Comment (zchsh) [inline src/content/home-page.json:12]: I've added `.` here for copy consistency with the HCP slot.

File: src/content/home-page.json [hunk 2]
```diff
@@ -9,11 +9,14 @@
...
 			"ctaText": "Learn more"
 		},
-		"hashiconfGlobal": {
-			"description": "Two days of inspirational speakers, community connection, and a front-row seat to future of cloud."
+		"hcp": {
+			"cardTitle": "HashiCorp Cloud Platform",
+			"url": "https://portal.cloud.hashicorp.com/sign-up",
```
Comment (zchsh) [inline src/content/home-page.json:17]: I think this is the right URL, based on this Asana task: 🎟️ [Decide on activity to highlight](https://app.asana.com/0/0/1203011399855183/f)

File: src/views/homepage/components/merchandising-slots/slots/hcp-slot/index.tsx [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import { LogoCtaSlot } from '../logo-cta-slot'
+import logo from '@hashicorp/mktg-logos/product/hcp/primary/black.svg?include'
+import { HcpSlotProps } from './types'
+import s from './hcp-slot.module.css'
+
+function HcpSlot({ url, cardTitle, description, ctaText }: HcpSlotProps) {
+	return (
+		<LogoCtaSlot
```
Comment (zchsh) [inline src/views/homepage/components/merchandising-slots/slots/hcp-slot/index.tsx:8]: I've set up a `LogoCtaSlot`. It handles the logo positioning and most other display concerns for both the HCP and Vault slots. The specific slots can customize background elements, logo SVG src, and description max-width.

Probably not the perfect abstraction, but felt like a good balance to meet our short-term needs here.

File: src/views/homepage/components/merchandising-slots/slots/vault-slot/index.tsx [hunk 4]
```diff
@@ -1,24 +1,18 @@
...
-				<Image src={logo} width={118} height={50} alt="" />
-			</div>
-			<Text className={s.description} weight="bold">
-				{description}
-			</Text>
-			<p className={s.cta}>
-				{ctaText} <IconArrowRight16 />
-			</p>
-		</CardLink>
+		<LogoCtaSlot
```
Comment (zchsh) [inline src/views/homepage/components/merchandising-slots/slots/vault-slot/index.tsx:8]: I've updated the Vault slot to use `LogoCtaSlot`, for consistency (see `Anything else` section in PR description).

File: src/views/homepage/index.tsx [hunk 5]
```diff
@@ -53,8 +50,11 @@ const HomePageContent = ({
 					description={merchandising.vault.description}
 					ctaText={merchandising.vault.ctaText}
 				/>
-				<HashiConfGlobalSlot
-					description={merchandising.hashiconfGlobal.description}
+				<HcpSlot
```
Comment (zchsh) [inline src/views/homepage/index.tsx:53]: Have replaced the `HashiConfGlobalSlot` with `HcpSlot`.

**Comment by zchsh** [comment]
@BRKalow @kellyschummer This PR updates the home page to display an HCP callout (replacing the previous HashiConf Global callout). Let me know if this looks alright! I think our aim is to ship this today 🤞 ⚡

**Review by BRKalow — APPROVED** [review]
LGTM!

**Review by kellyschummer — APPROVED** [review]
Thank you @zchsh! Looking great!

**Pull request merged**